### PR TITLE
Addlog

### PR DESCRIPTION
**Pull request by lucusfly**

### CODE REVIEW

File: log/log.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    "fmt"
+    "github.com/sirupsen/logrus"
+    "os"
+    "os/user"
+    "path/filepath"
+)
+
+var log *logrus.Logger
+
+// PanicLevel level, highest level of severity. Logs and then calls panic with the
```
Comment (WTer) [inline log/log.go]: nice to add a space between Panic and Level:
PanicLevel  --> Panic Level

to meet golint format requirement.

File: log/log.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
+/**
+*  @file
+*  @copyright defined in go-seele/LICENSE
+ */
+package log
```
Comment (WTer) [inline log/log.go]: add unittests for this package.

File: log/log.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    "os"
+    "os/user"
+    "path/filepath"
+)
+
+var log *logrus.Logger
+
+// PanicLevel level, highest level of severity. Logs and then calls panic with the
+// message passed to Debug, Info, ...
+func Panic(args ...interface{}) {
```
Comment (WTer) [inline log/log.go]: all these functions are thread-safe?

File: log/log.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+// DebugLevel level. Usually only enabled when debugging. Very verbose logging.
+func Debug(args ...interface{}) {
+    log.Debugln(args)
+}
+
+func init()  {
+    usr, err := user.Current()
+    if err != nil {
```
Comment (WTer) [inline log/log.go]: why do we get current user and set it as the required condition for log initializaiton?

File: log/log.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    seeleFolder := filepath.Join(usr.HomeDir, "seelelog")
+    if _, err := os.Stat(seeleFolder); err != nil {
+        if err = os.Mkdir(seeleFolder, os.ModeDir); err != nil {
+            fmt.Println("create log folder failed. ", err)
+            return
+        }
+    }
+
+    seeleLog := filepath.Join(seeleFolder, "log.txt")
+    file, err := os.OpenFile(seeleLog, os.O_CREATE | os.O_WRONLY, 0666)
```
Comment (WTer) [inline log/log.go]: what's happened when this function called multiple times by the same user?

File: log/log.go (hunk 2 above)
Comment (lucusfly) [reply log/log.go]: add log_test

File: log/log.go (hunk 1 above)
Comment (lucusfly) [reply log/log.go]: sure

File: log/log.go (hunk 3 above)
Comment (lucusfly) [reply log/log.go]: yes, logrus are thread-safe

File: log/log.go (hunk 4 above)
Comment (lucusfly) [reply log/log.go]: we need find a log file, by default I use $USER/seelelog/log.txt as log file, maybe we could have a better way to find it later

File: log/log.go (hunk 5 above)
Comment (lucusfly) [reply log/log.go]: the init function is called automatically when package initialized. no one should call it.

**Pull request merged**